src/main/kotlin/net/djvk/fireflyPlaidConnector2/transactions/PlaidFireflyTransaction.kt

```diff
@@ -0,0 +1,127 @@
+package net.djvk.fireflyPlaidConnector2.transactions
+
+import net.djvk.fireflyPlaidConnector2.api.firefly.models.TransactionTypeProperty
+import net.djvk.fireflyPlaidConnector2.api.plaid.models.Transaction
+import java.time.OffsetDateTime
+import java.time.ZoneId
+
+/**
+ * Represents a single non-transfer transaction, including both the Plaid and the Firefly representation of it.
+ */
+sealed interface PlaidFireflyTransaction {
+    val amount: Double
+    val transactionId: String
+    val fireflyAccountId: Int
+    val fireflyTransaction: FireflyTransactionDto?
+    val plaidTransaction: Transaction?
+    fun getTimestamp(zoneId: ZoneId): OffsetDateTime
+
+    companion object {
+        /**
+         * Wraps Plaid and Firefly transactions into PlaidFireflyTransaction objects, joining them together when
+         * matching transactions are identified.
```

Comment:
"Matching" meaning what? Identical ids? This is probably worth clarifying as it overlaps a bit semantically with the transfer matching concept.

Reply:
Renamed `match` to `normalizeByTransactionId` and updated documentation accordingly in https://github.com/nprzy/firefly-plaid-connector-2/commit/a1fc26cbe9641b94e500301e37b4d24adba14d54